finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/TlsClientPrep.scala

```diff
@@ -0,0 +1,203 @@
+package com.twitter.finagle.buoyant
+
+import com.twitter.finagle._
+import com.twitter.finagle.client.Transporter
+import com.twitter.finagle.ssl.{Engine, Ssl}
+import com.twitter.finagle.transport.{TlsConfig, Transport}
+import java.io.FileInputStream
+import java.net.{InetSocketAddress, SocketAddress}
+import java.security.KeyStore
+import java.security.cert.{CertificateFactory, X509Certificate}
+import javax.net.ssl.{SSLContext, TrustManagerFactory}
+
+object TlsClientPrep {
+
+  object role extends Stack.Role("TlsClientPrep") {
+    val finagle = Stack.Role("TlsClientPrep.Finagle")
+  }
+
+  val description = "Configures per-endpoint TLS settings"
+
+  type Params = Stack.Params
+
+  /**
+   * Configures TLS protocol parameters, including whether TLS should
+   * be used at all.
+   */
+  case class TransportSecurity(config: TransportSecurity.Config)
+  implicit object TransportSecurity extends Stack.Param[TransportSecurity] {
+    sealed trait Config
+
+    /** TODO support Protocols, Ciphers, & timeouts */
+    case class Secure() extends Config
+
+    object Insecure extends Config
```

Comment:
Note: I moved TlsEnabled to TransportSecurity, so this can later include tls protocol & cipher configuration.

I moved to wrapper types (rather than making the param an ADT) because it simplifies type inference for params (also this tends to be what we see in finagle)